tests/api/v1/collectors/heartbeat.js

```diff
@@ -1017,7 +1018,43 @@ describe('tests/api/v1/collectors/heartbeat.js >', () => {
         .then(done)
         .catch(done);
       });
+    });
+
+    describe('added generator >', () => {
+      before(() => tu.toggleOverride('returnUser', true));
+      after(() => tu.toggleOverride('returnUser', false));
+
+      // setup and create a new generator with the createdBy field
+      it('contains the user token', (done) => {
+        const gtPath = '/v1/generatorTemplates';
+        const sgtCopy = JSON.parse(JSON.stringify(sgt));
+        sgtCopy.name = 'iAmNew';
+        sgtCopy.createdBy = userId;
+        sgtCopy.contextDefinition = contextDefinition;
+        gu.createSGtoSGTMapping(sgtCopy, generator2);
+
+        api.post(gtPath)
+        .set('Authorization', userToken)
+        .send(sgtCopy)
+        .expect(constants.httpStatus.CREATED)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
 
+          expect(res.body.createdBy).to.equal(userId);
+          const timestamp = Date.now();
+          return Promise.resolve()
+          .then(() => postGenerator(generator2, [collector1]))
+          .then(() => sendHeartbeat(collector1))
+          .then((res) => {
+            const reencryptedSG = res.body.generatorsAdded[0];
+            expect(reencryptedSG.token).to.be.an('string');
```

Comment:
shouldn't you also check that the value of the token is correct?

Reply:
do you mean check it is a proper UUID, or the expected UUID? The latter is difficult as the jwtUtil token generation is one way.

Reply:
ok, makes sense.

Reply:
@jgraff2 how do I check the value of the token is correct? Do I use jwtUtil.verifyToken(req, callback)? or make a new function in jwtUtil with 
  return jwtVerifyAsync(token, secret, {})
  .then((payLoad) => {
    decodedData = payLoad;
    return User.findOne({ where: { name: decodedData.username } });
  })
  .then((user) => {
    if (!user) {
      throw new apiErrors.ForbiddenError({
        explanation: 'Invalid Token.',
      });
    }

    req.user = user.get();

Reply:
@jgraff2 in retrospect that is a lot of work to check it is a token. Any suggestions?

Reply:
I think it's fine as-is, given that the token is being created just before this and createToken() is tested elsewhere.